docs/articles/documentation/using-testcafe/common-concepts/inject-scripts-into-tested-pages.md

```diff
@@ -0,0 +1,364 @@
+---
+layout: docs
+title: Inject Scripts into Tested Pages
+permalink: /documentation/using-testcafe/common-concepts/inject-scripts-into-tested-pages.html
+---
+# Inject Scripts into Tested Pages
+
+TestCafe allows you to inject custom client scripts into pages visited during the tests. You can add scripts that mock browser API or provide helper functions.
+
+Use [test run options](#add-client-scripts-to-all-tests) to add client scripts to all tests, or [test API](#add-client-scripts-to-specific-tests) to add them to specific fixtures or tests.
+
+## Add Client Scripts to All Tests
+
+Use either of the following to inject scripts into pages visited during all the tests:
+
+* the [--cs (--client-scripts)](../command-line-interface.md#--cs-pathpath2---client-scripts-pathpath2) command line option
+
+    ```sh
+    testcafe chrome test.js --client-scripts=mockDate.js,assets/react-helpers.js
+    ```
+
+* the [runner.clientScripts](../programming-interface/runner.md#clientscripts) API method
+
+    ```js
+    runner.clientScripts('mockDate.js', 'scripts/react-helpers.js');
+    ```
+
+* the [clientScripts](../configuration-file.md#clientscripts) configuration file property
+
+    ```json
+    {
+        "clientScripts": ["mockDate.js", "scripts/react-helpers.js"]
+    }
+    ```
+
+## Add Client Scripts to Specific Tests
+
+Use the [fixture.clientScripts](../../test-api/test-code-structure.md#inject-scripts-into-tested-pages) and [test.clientScripts](../../test-api/test-code-structure.md#inject-scripts-into-tested-pages) methods to inject scripts into pages visited during a particular test or fixture.
+
+```js
+fixture `My fixture`
+    .page `http://example.com`
+    .clientScripts('assets/jquery.js');
+```
+
+```js
+test
+    ('My test', async t => { /* ... */ })
+    .clientScripts({ module: 'async' });
+```
+
+## Provide Scripts to Inject
+
+You can pass the following arguments to specify the scripts to inject:
+
+* [path to a JavaScript file](#inject-a-javascript-file)
+* [module name](#inject-a-module)
+* [script code](#inject-script-code)
+
+You can also [inject scripts into specific pages](#provide-scripts-for-specific-pages).
+
+> Note that the API methods and configuration option support [multiple arguments](#specify-multiple-scripts).
+
+### Inject a JavaScript File
+
+Specify the JavaScript file path to inject the entire content of this file into the tested pages. You can pass a string or an object with the `path` property.
+
+*CLI*
+
+```text
+testcafe <browser> <tests> --cs <path>
+```
+
+*Programming interface*
+
+```text
+runner.clientScripts( path | { path } )
```

Comment:
According to https://github.com/DevExpress/testcafe/blob/2040ad57ccf8a31375e48cd6da93653a39c981d5/docs/articles/documentation/using-testcafe/programming-interface/runner.md#clientscripts
a path can be an array as well.

Reply:
The `runner.clientScripts` method can take arrays. This is reflected in its `script` parameter: https://github.com/VasilyStrelyaev/testcafe/blob/add-client-scripts/docs/articles/documentation/using-testcafe/programming-interface/runner.md#clientscripts

But here we're talking about `path` specifically, either in a string or as a part of an object. And in general, it can't be an array. You can't do:

```js
runner.clientScripts({ path: ['path1.js', 'path2.js'] });
```

Reply:
Let's discuss it in private, because it looks too cryptic for me now.
Anyway, I can visit documentation from any page and it should look clear on each page.